dart/lib/src/types.dart

```diff
@@ -0,0 +1,152 @@
+import 'dart:typed_data';
+
+enum BitWidth {
+  width8,
+  width16,
+  width32,
+  width64
+}
+
+class BitWidthUtil {
```

Comment:
Document all public types.

And does this really need to be public?

Also, I'm pretty sure at least some of this will fail when converted to JavaScript in Flutter for Web/dart2js.

Reply:
The Util classes are just "static" classes with static methods, which could be private if we would merge all 3 files into one. I also realised that Dart sports top level function definition, so we don't need classes at all.

Could you please elaborate on your concerns regarding Web/dart2js and would it be resolved if I would move the methods into top level functions.

Also what are your thoughts regarding single file vs. `part of` (https://dart.dev/guides/language/effective-dart/usage#do-use-strings-in-part-of-directives).

Reply:
For visibility: If this is public, someone will use it, and depend on it, and we will end up breaking them if we change it, even to fix bugs. These all seem like internal private utility methods. It would probably be better to move them to another file in the library that is _not_ exported. That way, we are making it a little clearer that these are internal API and not meant to be used by developers. 
The guidance from the Dart team is basically to avoid using `part` files if you can. 


For dart2js: In JavaScript, numbers are capped at 53 bits. This class has operations expecting to be able to work with 64 bit numbers.

Reply:
Oh! This file isn't exported. Nevermind :)

Reply:
So as far as I understand as the files are not exported in flexbuffers.dart we are good :).

Resolving this one.